modules/apps/transfer/keeper/msg_server.go

```diff
@@ -59,3 +66,33 @@ func (k Keeper) UpdateParams(goCtx context.Context, msg *types.MsgUpdateParams)
 
 	return &types.MsgUpdateParamsResponse{}, nil
 }
+
+// unwindHops unwinds the hops present in the tokens denomination and returns the message modified to reflect
+// the unwound path to take. It ensures only a single token must be present in the tokens list
+// and that the token is not native to the chain.
+func (k Keeper) unwindHops(ctx sdk.Context, msg *types.MsgTransfer) (*types.MsgTransfer, error) {
+	coins := msg.GetCoins()
+	if len(coins) > 1 {
+		return nil, errorsmod.Wrap(types.ErrInvalidForwarding, "cannot unwind more that one tokens")
+	}
+
+	token, err := k.tokenFromCoin(ctx, coins[0])
+	if err != nil {
+		return nil, err
+	}
+
+	if token.Denom.IsNative() {
+		return nil, errorsmod.Wrap(types.ErrInvalidForwarding, "cannot unwind a native token")
+	}
```

Comment:
Should we also move this to `ValidateBasic`?

Reply:
I wasn't able to achieve it since it's consume states from the keeper (`k.tokenFromCoin`) open to suggestions on how to implement it if it's possible though

Reply:
Ah, yes, that's true.